deeprankcore/dataset.py

```diff
@@ -658,17 +658,43 @@ def __init__( # pylint: disable=too-many-arguments, too-many-locals
         if not train:
             if not isinstance(dataset_train, GraphDataset):
                 raise TypeError("Please provide a valid training GraphDataset.")
-
+            
+            #set features_transform as the same in dataset_train
             if dataset_train.features_transform is not None:
                 self.features_transform = dataset_train.features_transform
             else:
                 self.features_transform = None
             _log.warning("features_transform parameter, if set, will be ignored." + "\n" +
-                        "features_transform will remain the same as the one used in training phase.")
+                        "features_transform will remain the same as the one used in training phase: %s.", self.features_transform)
+
+            #set task as the same in dataset_train
+            self.task = dataset_train.task
+            _log.warning("task parameter, if set, will be ignored." + "\n" +
+                        "task will remain the same as the one used in training phase: %s.", self.task)            
```

Comment:
I feel that we only need to add these warnings (or maybe even throw errors?) in case a user set the parameters manually and they differ from the inherited one.
If that doesn't happen, I don't see any reason why this needs to go into the log or warnings or anywhere. It is the default expected behavior of the software.
One way to do this is to loop through the different inherited settings and then check for each whether it matches the user set one and deal accordingly.

Also, i personally very much prefer f-strings over %-formatting (plus I think it's the [recommended way](https://realpython.com/python-f-strings/) now).

Reply:
I don't think I understand fully what you mentioned about `dataset_train`.
Do you mean to move its order in the definition before the parameters that needed to be inherited?
Besides, I modified the docstring to state the parameters that will be ignored and inherited.